superset-frontend/src/explore/components/controls/DateFilterControl/AdvancedFrame.tsx

```diff
@@ -0,0 +1,70 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { t } from '@superset-ui/core';
+import { SEPARATOR } from 'src/explore/dateFilterUtils';
+import { Input } from 'src/common/components';
+
+type AdvancedFrameProps = {
+  onChange: (timeRange: string) => void;
+  value: string;
+};
```

Comment:
Nit: for components with very few props, maybe you can just inline the typing and use object deconstrution.

```ts
export default function AdvancedFrame({ onChange, value }: {
  onChange: (timeRange: string) => void;
  value: string;
}) {
```

However, I think all the different subcomponents would have the same signature, so you should probably reuse the `Props` type. E.g. add a `FrameComponentProps` to `types.ts`.

I'd also recommend create another subfolder to make the file directories easier to glance.

Reply:
Thank you for the reminder.  I will change this typing to inline.